engines/onboarding/app/services/onboarding/invitation_service.rb

```diff
@@ -1,40 +1,71 @@
 module Onboarding
   class InvitationService
-    attr_accessor :invitation, :creator
 
-    # @param invitation [Invitation]
-    def initialize(invitation)
-      @invitation = invitation
+    def initialize
     end
 
-    # Persists the Invitation
-    # and enqueues an InvitationJob
+    # Persists the Invitation and enqueues an InvitationJob
     #
-    # @return [Boolean]
-    def create
-      if @invitation.save
-        send_invitation_email
+    # @param email [String]
+    # @param group [Group]
+    # @param inviting_user [User]
+    # @return [Invitation]
+    def create!(email, group, inviting_user)
+      invitation = load_invitation(email, group, inviting_user)
 
-        true
+      if invitation.save
+        # TODO: move to background job
+        # invitation.update_attribute(sent_at: Time.zone.utc.now)
+        InvitationMailer.invite_user(invitation).deliver
       end
+
+      invitation
+    end
+
+    # Enqueues a job to create a bunch of invitations
+    #
+    # @param group [Group]
+    # @param inviting_user [User]
+    # @param emails [Array<String>]
+    def bulk_invite!(group, inviting_user, emails)
+      # TODO: move to background job
+      emails.each { |email| create!(email, group, inviting_user) }
     end
 
     # Accepts an invitation
     # a new user is created and added to the group which has been invited
-    def accept!
-      add_user_to_group
+    #
+    # @param signup [Account::Signup]
+    # @param options [Hash]
+    # @option options [String] :username
+    # @option options [String] :password
+    # @option options [String] :password_confirmation
+    # @option options [String] :first_name
+    # @option options [String] :last_name
+    # @return [Account::User]
+    def accept!(invitation, options)
```

Comment:
this is wrong, we should:
 - create the user only if not already existent
 - add him as participant to the group